src/rebuild/initiator.c

```diff
@@ -353,6 +354,14 @@ rebuild_dkey(struct rebuild_tgt_pool_tracker *rpt,
 			rc = rebuild_fetch_update_bulk(rdone, oh, rebuild_cont);
 	}
 
+	if (rc == -DER_INPROGRESS) {
```

Comment:
Who does return the -DER_INPROGRESS, the vos_update_end()? I think that the vos_update_end() tries to update the local vos with the fetched data from leader replica, right? If yes, the local vos should NOT have any conflict with the in-rebuilding target. There may be something wrong.

Reply:
not sure about vos_update_end(), but seems ds_obj_fetch() may get -DER_INPROGRESS

Reply:
For rebuild case, we need to pass the DAOS_INTENT_REBUILD intent into VOS, then there will be no -DER_INPROGRESS when fetch record with non-committed DTX.

Reply:
rebuild is using normal i/o stack to fetch data from client side, hmm, not easy to pass this intent through the whole stack, unless we add some specialities into the I/O stack for rebuild, which I really do not want.
Btw: what is the difference between rebuild intent fetch vs normal fetch?

Reply:
If the fetch on leader hits non-committed DTX, it will get -DER_INPROGRESS, but for rebuild case, it will skip such non-committed DTX without -DER_INPROGRESS. Because we assume that: before the object rebuild start, all former exist DTXs are either been committed or aborted, so the DTX with 'prepared' status must be for new update after the object rebuild start. Such new update should have also happened on the new target, so it is unnecessary to make such update to be rebuilt.